src/contract/assembled_transaction.ts

```diff
@@ -405,17 +405,21 @@ export class AssembledTransaction<T> {
     }
     const method = invokeContractArgs.functionName().toString('utf-8');
     const txn = new AssembledTransaction(
-      { ...options,
+      {
+        ...options,
         method,
         parseResultXdr: (result: xdr.ScVal) =>
           spec.funcResToNative(method, result),
       }
-     );
+    );
     txn.built = built;
     return txn;
   }
 
   private constructor(public options: AssembledTransactionOptions<T>) {
+    if (!options.publicKey) {
+      throw new Error("Public key not provided. Have you forgotten to set the `publicKey` in AssembledTransactionOptions?");
```

Comment:
@chadoh correct me if I'm wrong, but doesn't `AssembledTransaction` still work without `sign` in the read-only case?